src/test/java/BaseTest.java

```diff
@@ -1,18 +1,142 @@
 import io.github.bonigarcia.wdm.WebDriverManager;
+import org.apache.commons.lang3.ObjectUtils;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.chrome.ChromeDriver;
-import org.openqa.selenium.firefox.FirefoxDriver;
-import org.openqa.selenium.safari.SafariDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.interactions.Action;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
 import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Parameters;
+
+import java.time.Duration;
 
 
 public class BaseTest {
+   public static WebDriver driver = null;
+   public static String url = null;
+   public static WebDriverWait wait = null;
 
+   public static Actions actions = null;
+  
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
+    @BeforeMethod
+    @Parameters({"baseUrl"})
+    public void launchBrowser(String baseUrl){
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--disable-notifications","--remote-allow-origins=*","--incognito","--start-maximize");
+
+        driver = new ChromeDriver(options);
+        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        driver.manage().window().maximize();
+        url = baseUrl;
+        driver.get(url);
+        
+    }
+
+    public static void provideEmail(String email){
+     WebElement emailField = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("[type='email']")));
+     emailField.click();
+     emailField.sendKeys(email);
+     
+    }
+      public static void providePassword(String password){
+        WebElement passwordField = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("[type='password']")));
+        passwordField.click();
+        passwordField.sendKeys(password);
+      }
+    
+      public static void clickSubmit(){
+        WebElement submitButton = wait.until(ExpectedConditions.elementToBeClickable
+                (By.cssSelector("[type='submit']")));
+        submitButton.click();
+  
+      }
+      public static void logIn(String email, String password){
+        provideEmail(email);
+        providePassword(password);
+        clickSubmit();
+        
+      }
+        public static void createNewPlaylist(){
+        WebElement newPlaylistIcon = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("i[title='create new playlist']")));
```

Comment:
You got this problem, please correct it. 

![image](https://user-images.githubusercontent.com/104761482/225846569-21a15e69-244e-4792-96f4-4032f5f0fcf1.png)

Reply:
Hello,
can you tell me how to fix this one? thank you!

Reply:
Hello

Please, re-submit homework 20 for review

Reply:
so I retest but kept getting this error, could you tell me what i did wrong before re-submitting? Thank you!
<img width="914" alt="image" src="https://user-images.githubusercontent.com/126042724/226451203-56bf7930-6dda-428e-8241-c5dd5927514a.png">

Reply:
Don't worry! Please, re -submit your homework20 and I'll check it.